Always create runtime-link using sharedlibs on aix.

Note that if you update this patch, please update this two as well,
each hunk in one file, as they may independently appear more than once
in configure scripts:

  eclass/ELT-patches/aixrtl/iftest
  eclass/ELT-patches/aixrtl/casetest

--- libtool.m4.orig	2007-04-03 10:48:46 +0200
+++ libtool.m4	2007-04-03 10:55:11 +0200
@@ -2929,14 +2929,8 @@
       # AIX style linking. If -brtl is somewhere in LDFLAGS, we
       # need to do runtime linking.
       case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	for ld_flag in $LDFLAGS; do
-	  case $ld_flag in
-	  *-brtl*)
-	    aix_use_runtimelinking=yes
-	    break
-	    ;;
-	  esac
-	done
+	# we do runtimelinking by default now.
+	aix_use_runtimelinking=no
 	;;
       esac
 
@@ -5830,12 +5824,8 @@
 	# AIX style linking. If -brtl is somewhere in LDFLAGS, we
 	# need to do runtime linking.
 	case $host_os in aix4.[[23]]|aix4.[[23]].*|aix5*)
-	  for ld_flag in $LDFLAGS; do
-  	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
-  	    aix_use_runtimelinking=yes
-  	    break
-  	  fi
-	  done
+	  # we use runtimelinking by default.
+  	  aix_use_runtimelinking=yes
 	  ;;
 	esac
 
